syntax = "proto3";

import "RaftMessage.proto";
import "Message.proto";

package pb;

message Packet {
    enum Type {
        PRESENCE = 0;
        ACK = 1;
        RAFT = 2;
        MESSAGE = 3; // TODO delete once raft works
        VOIP = 4; // bypass Raft
    }

    Type message_type = 1;
    uint32 sequence_number = 2;
    uint32 acknowledgment_number = 3;
    uint32 sender_ip = 4;
    repeated uint32 receiver_ip = 5;
    uint32 ttl = 6; // hop count, probably 1 or 2

    RaftMessage raft_msg = 7;
    // Checksum of the message
    bytes checksum = 8;
    
    // once raft works, the raft message will contain
    // the actual message. Until then, we can bypass it.
    Message msg = 9; 

    // Audio data should not go trhough Raft
    bytes audio = 10;
}
